coffee-chats/src/main/webapp/src/components/FindChatCard.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState, ChangeEvent } from "react"
+import "./FindChatCard.css"
+import { Typography, CardActions, Card, CardContent, Button, CardHeader, createStyles, makeStyles, Theme, Collapse, Grid, MenuItem, InputLabel, Select, FormControl, FormControlLabel, Checkbox, CircularProgress } from "@material-ui/core"
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore"
+import clsx from "clsx";
+import { addWeeks, startOfWeek, addDays } from "date-fns";
+import { MultiDatePicker } from "./MultiDatePicker";
+import { green } from "@material-ui/core/colors";
+import { submitChatRequest } from "../utils/chatRequest";
+
+const useStyles = makeStyles((theme: Theme) => 
+  createStyles({
+    expand: {
+      transform: 'rotate(0deg)',
+      transition: theme.transitions.create('transform', {
+        duration: theme.transitions.duration.shortest,
+      }),
+    },
+    expandOpen: {
+      transform: 'rotate(180deg)',
+    },
+    actions: {
+      display: 'flex',
+      justifyContent: 'space-between'
+    },
+    successBtnWrapper: {
+      margin: theme.spacing(1),
+      position: 'relative',
+    },
+    btnSuccess: {
+      backgroundColor: green[500],
+      '&:hover': {
+        backgroundColor: green[700]
+      }
+    },
+    btnProgress: {
+      color: green[500],
+      position: 'absolute',
+      top: '50%',
+      left: '50%',
+      marginTop: -12,
+      marginLeft: -12,
+    }
+  })
+);
+
+interface FindChatCardProps {
+  interests: string[],
+}
+
+export const FindChatCard: React.FC<FindChatCardProps> = ({ interests }) => {
+  
+  const classes = useStyles();
+
+  const MONDAY = 1;
+  const startOfNextWeek = startOfWeek(addWeeks(new Date(), 1), {weekStartsOn: MONDAY});
+
+  const [expanded, setExpanded] = useState(false);
+  const [dates, setDates] = useState(Array.from(Array(5).keys()).map((i: number) => addDays(startOfNextWeek, i)))
+  const [numPeople, setNumPeople] = useState(1);
+  const [duration, setDuration] = useState(30);
+  const [randomMatchChecked, setRandomMatchChecked] = useState(false);
+  const [pastMatched, setPastMatched] = useState(true);
+
+  const [loading, setLoading] = React.useState(false);
+  const [success, setSuccess] = React.useState(false);
+
+  const handleExpandClick = () => {
+    setExpanded(!expanded);
+  }
+
+  const chatRequestClick = async () => {
+    if (!loading) {
+      setSuccess(false);
+      setLoading(true);
+
+      setSuccess(await submitChatRequest(interests, dates, numPeople, duration, randomMatchChecked, pastMatched));
+      setLoading(false);
+    }
+  }
+
+  return (
+    <Card>
+      <CardHeader title="Find a chat" />
+      <CardContent>
+        <Typography>
+          Find someone to chat to about {(interests.length > 0) ? interests.join(", ") : 'anything'}.
```

Comment:
I've made a `humanReadableJoin`, it lives in `format.ts` and might be useful here (though it responds "X, Y and Z", where here "X, Y or Z" is probably expected)

Reply:
I did try to use this function, however the function does not work correctly with 'state' variables as it attempts to modify the actual array, which began to introduce massive inconsistencies in the state. At the moment this is just a placeholder for when the tags and chips are added, so for now I think it works okay.